### DISCUSSION TITLE
Handle Character

### DISCUSSION THREAD

**Discussion by NiranjanShah** [bosskmk/pluto_grid#203] [General]
In PlutoGridKeyManager there is a function _handleCharacter which is as below, **there must be a condition only to _handleCharacter if the grid is in edit, Currently even though the grid is in SelectMode, this event is taking place**

void _handleCharacter(PlutoKeyManagerEvent plutoKeyManagerEvent) {
   **below if statement must be added (as per me
if (stateManager.selectingMode == PlutoGridSelectingMode.none){
      return;
    }
**

    if (stateManager!.isEditing != true && stateManager!.currentCell != null) {
      stateManager!.setEditing(true);

      stateManager!.changeCellValue(stateManager!.currentCell!.key,
          plutoKeyManagerEvent.event!.logicalKey.keyLabel);
    }
  }

**Comment by bosskmk** [comment]
selectingMode has cell, row, none, and this has nothing to do with editing mode. A condition to select a row.
What has to do with the edit mode is the normal, select of PlutoGridMode . 
When creating a grid, if the mode is set to select, editing is not possible and only row selection is possible.

**Reply by NiranjanShah** [reply]
however, if grid is in Edit mode and focus is not on Grid then ?
